metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py

```diff
@@ -477,6 +479,20 @@ def validate_shares(
 
         return shares
 
+    @root_validator(pre=False, skip_on_failure=True)
+    def validate_queries_v2_stateful_ingestion(cls, values: Dict) -> Dict:
```

Comment:
FYI I'm replacing these v1 validators in https://github.com/datahub-project/datahub/pull/15057